File: app/src/main/java/edu/application/MyApplication.java
package edu.application;

import android.annotation.SuppressLint;
import android.app.Application;
import android.content.Context;

import com.google.firebase.FirebaseApp;

public class MyApplication extends Application {

    @SuppressLint("StaticFieldLeak")
    private static Context context;

    public void onCreate() {
        super.onCreate();
        MyApplication.context = getApplicationContext();
        FirebaseApp.initializeApp(this);
    }

    public static Context getAppContext() {
        return MyApplication.context;
    }
}

File: app/src/main/kotlin/edu/application/MyApplication.kt
package edu.application

import android.app.Application
import android.content.Context
import com.google.firebase.FirebaseApp

class MyApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        appContext = applicationContext

        FirebaseApp.initializeApp(this)
    }

    companion object {
        lateinit var appContext: Context
            private set
            @JvmStatic get
    }
}